cli/src/main/kotlin/tools/samt/cli/DiagnosticFormatter.kt

```diff
@@ -0,0 +1,505 @@
+        append(formatTextForSeverity(lastRowHighlightedPortion, severity, withBackground = true))
+        appendLine(gray(lastRowNonHighlightedPortion))
+
+        // print optional highlight message
+        if (highlight.message != null) {
+            appendLine(formatHighlightedMultilineEmptySection(severity))
+            append(formatTextForSeverity("+--------- ", severity))
+            appendLine(formatTextForSeverity(highlight.message!!, severity))
+            appendLine(formatNonHighlightedEmptySection())
+        }
+    }
+
+    private fun formatHighlightAnnotations(annotations: List<DiagnosticAnnotation>): String = buildString {
+        require(annotations.isNotEmpty())
+
+        for (annotation in annotations) {
+            when (annotation) {
+                is DiagnosticAnnotationHelp -> {
+                    append("        = ")
+                    append(green("help:"))
+                    append(" ")
+                    appendLine(formatHighlightAnnotationsMessage(annotation.message, indentStart = 16))
+                }
+                is DiagnosticAnnotationInformation -> {
+                    append("        = ")
+                    append(blue("info:"))
+                    append(" ")
+                    appendLine(formatHighlightAnnotationsMessage(annotation.message, indentStart = 16))
+                }
+            }
+        }
+    }
+
+    private fun formatHighlightAnnotationsMessage(message: String, indentStart: Int): String = buildString {
+        val maxLineLength = (terminalWidth / 3) * 2
+
+        // message fits into terminal width
+        if (message.length <= maxLineLength) {
+            append(message)
+            return@buildString
+        }
+
+        // split message into multiple lines each no longer than the max line length
+        val lines = message.splitToSequence(" ")
+            .fold(mutableListOf<String>()) { lines, word ->
+                if (lines.isEmpty()) {
+                    lines.add(word)
+                } else {
+                    val lastLine = lines.last()
+                    if (lastLine.length + word.length + 1 <= maxLineLength) {
+                        lines[lines.lastIndex] = "$lastLine $word"
+                    } else {
+                        lines.add(word)
+                    }
+                }
+                lines
+            }
+
+        lines.forEachIndexed { index, line ->
+            append(line)
+
+            if (index != lines.lastIndex) {
+                appendLine()
+                append(" ".repeat(indentStart))
+            }
+        }
+
+    }
+
+    private fun formatNonHighlightedEmptySection(): String = buildString {
+        append("        ┃ ")
+    }
+
+    private fun formatHighlightedLineNumberSection(row: Int, severity: DiagnosticSeverity): String = buildString {
+        append("   ")
+        append(formatTextForSeverity((row + 1).toString().padStart(4), severity, withBold = true))
+        append(" ┃ ")
```

Comment:
This line has too much girth for the Windows terminal, maybe replace them all with the less thick `│`?